src/app/ui/editor/tool_loop_impl.cpp

```diff
@@ -973,9 +973,12 @@ class PreviewToolLoopImpl : public ToolLoopBase {
         tools::WellKnownPointShapes::Brush);
     }
     else if (m_pointShape->isFloodFill()) {
-      m_pointShape = App::instance()->toolBox()->getPointShapeById
-        (m_tilesMode ? tools::WellKnownPointShapes::Tile:
-                       tools::WellKnownPointShapes::Pixel);
+      if (m_tilesMode)
+        m_pointShape = App::instance()->toolBox()->getPointShapeById(tools::WellKnownPointShapes::Tile);
+      else if (m_brush->type() == BrushType::kImageBrushType)
+        m_pointShape = App::instance()->toolBox()->getPointShapeById(tools::WellKnownPointShapes::Brush);
+      else
+        m_pointShape = App::instance()->toolBox()->getPointShapeById(tools::WellKnownPointShapes::Pixel);
```

Comment:
Instead of calling `App::instance()->toolBox()->getPointShapeById()` you can assign the id into a variable and use it as argument in one call site
```suggestion
      const char* id;
      if (m_tilesMode)
        id = tools::WellKnownPointShapes::Tile;
      else if (m_brush->type() == BrushType::kImageBrushType)
        id = tools::WellKnownPointShapes::Brush;
      else
        id = tools::WellKnownPointShapes::Pixel;
      m_pointShape = App::instance()->toolBox()->getPointShapeById(id);
```
This is just to avoid writing `App::instance()->toolBox()->getPointShapeById` 3 times.